src/hooks/api/useCashFlow.ts

```diff
@@ -4,44 +4,71 @@ import type { UseQueryResult } from '@tanstack/react-query';
+import Money from '@/book/Money';
+
+export type CashFlowRow = {
+  guid: string,
+  total: Money,
+  type: string,
+  name: string,
+};
 
 /**
- * Retrieve the cashflow for a given account in the specified date.
+ * Retrieve the cashflow for a given account in the specified period.
  *
- * Returns an array containing the account guid, the total for that account, the type and
- * the name.
+ * The query works for transactions in multiple splits and in different currencies.
+ * We do that by calculating the exchange rate in the transaction and applying it to
+ * the splits for the other accounts.
  */
 export function useCashFlow(
   account: string,
   selectedInterval?: Interval,
-): UseQueryResult<{ guid: string, total: number, type: string, name: string }[]> {
+): UseQueryResult<CashFlowRow[]> {
   const { data: defaultInterval } = useInterval();
   const interval = selectedInterval || defaultInterval;
 
   const result = useQuery({
     queryKey: [...Split.CACHE_KEY, account, 'cashflow', interval.toISODate()],
-    queryFn: () => Split.query(`
-      SELECT
-        splits.account_guid as guid,
-        accounts.name,
-        accounts.account_type as type,
-        SUM(cast(splits.quantity_num as REAL) / splits.quantity_denom) as total
-      FROM transactions AS tx
-      JOIN splits ON splits.tx_guid = tx.guid
-      JOIN accounts ON splits.account_guid = accounts.guid
-      WHERE tx.guid IN (
-        SELECT DISTINCT tx_guid
-        FROM splits
-        WHERE account_guid = '${account}'
-      )
-      AND tx.post_date >= '${((interval as Interval).start as DateTime).toSQLDate()}'
-      AND tx.post_date <= '${((interval as Interval).end as DateTime).toSQLDate()}'
-      GROUP BY splits.account_guid
-      HAVING SUM(cast(splits.quantity_num as REAL) / splits.quantity_denom) != 0
-      ORDER BY total
-    `),
+    queryFn: async () => {
+      const rows = await Split.query(`
+        SELECT
+          splits.account_guid as guid,
+          accounts.name as name,
+          accounts.account_type as type,
+          (
+            SELECT commodities.mnemonic 
+            FROM commodities
+            JOIN accounts AS main_account ON main_account.guid = '${account}'
+            WHERE commodities.guid = main_account.commodity_guid
+          ) AS mnemonic,
+          SUM(((cast(main_split.quantity_num as REAL) / main_split.quantity_denom) / (cast(main_split.value_num as REAL) / main_split.value_denom)) * (cast(splits.value_num as REAL) / splits.value_denom)) as total
+
+        FROM transactions AS tx
+        JOIN splits ON splits.tx_guid = tx.guid
+        JOIN accounts ON splits.account_guid = accounts.guid
+        JOIN splits AS main_split ON main_split.tx_guid = tx.guid AND main_split.account_guid = '${account}'
+
+        WHERE tx.guid IN (
+          SELECT DISTINCT tx_guid
+          FROM splits
+          WHERE account_guid = '${account}'
+        )
+        AND accounts.guid != '${account}'
```

Comment:
This avoids having an extra row for the account we generate the report for. We could add this as a double check to check that the quantities match but no use case for now.